src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -66,6 +68,27 @@ public KestrelHttpParser(IKestrelTrace log)
                 span = startLineBuffer.ToSpan();
             }
 
+            // pathStart
+            // |    pathEnd and queryStart
+            // |    |       queryEnd
+            // |    |       |versionStart
+            // |    |       ||
+            // /path?query=1 HTTP/1.1
+
+            // pathStart
+            // |               pathEnd and queryStart
+            // |               |
+            // |               |       queryEnd
+            // |               |       |versionStart
+            // |               |       ||
+            // http://host/path?query=1 HTTP/1.1
```

Comment:
`pathStart` should be at the first `/` in the actual path. For the special cases `http://host` and `http://host?query`, make a span with `/` and pass that to `OnStartLine`.